### PR TITLE
Add aten::avg_pool2d/avg_pool2d_backward

### PR DESCRIPTION
**Pull request by chunhuanMeng**
ops need to support in this pr
- [x] avg_pool2d
- [x] avg_pool2d_backward

### CODE REVIEW

**Comment by chunhuanMeng** [comment]
image

File: yaml/xpu_functions.yaml [hunk 1]
```diff
@@ -386,6 +386,10 @@ supported:
   - acosh.out
   - addr
   - addr.out
+  - avg_pool2d
```
Comment (dvrogozh) [inline yaml/xpu_functions.yaml:389]: Will this get `aten::_adaptive_avg_pool2d` supported? This one is required for efficientnet inference.

File: src/ATen/native/xpu/AveragePool2d.cpp [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+        input.options().memory_format(memory_format));
+  } else {
+    grad_input = at::xpu::create_out(
+        input.sizes(), {}, input.options().memory_format(memory_format));
+  }
+  return grad_input;
+}
+
+Tensor XPUNativeFunctions::avg_pool2d(
+    const at::Tensor& input,
```
Comment (majing921201) [inline src/ATen/native/xpu/AveragePool2d.cpp:184]: we don't need "at::" in this file

File: src/ATen/native/xpu/sycl/AveragePool2dKernels.cpp [hunk 3]
```diff
@@ -0,0 +1,845 @@
...
+    Tensor& output,
+    const int divisor_override,
+    const bool count_include_pad,
+    const bool use_divisor) {
+  scalar_t* top_data = output.data_ptr<scalar_t>();
+  const scalar_t* bottom_data = input.data_ptr<scalar_t>();
+
+  auto& queue = at::xpu::getCurrentSYCLQueue();
+  const uint32_t group_size =
+      std::min(static_cast<int>(syclMaxWorkItemsPerEU()), 1024);
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/AveragePool2dKernels.cpp:251]: For sycl kernel, do we still need 1024 wg size limitation ?

File: src/ATen/native/xpu/sycl/AveragePool2dKernels.cpp [hunk 4]
```diff
@@ -0,0 +1,845 @@
...
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/AveragePool2dKernels.cpp:299]: ditto

File: src/ATen/native/xpu/sycl/AveragePool2dKernels.cpp [hunk 5]
```diff
@@ -0,0 +1,845 @@
...
+        kernel_w(kernel_w_),
+        stride_h(stride_h_),
+        stride_w(stride_w_),
+        pad_h(pad_h_),
+        pad_w(pad_w_),
+        divisor_override(divisor_override_),
+        count_include_pad(count_include_pad_),
+        use_divisor(use_divisor_) {}
+
+ private:
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/AveragePool2dKernels.cpp:408]: private variable naming with "_"

File: src/ATen/native/xpu/sycl/AveragePool2dKernels.cpp [hunk 6]
```diff
@@ -0,0 +1,845 @@
...
+    Tensor& grad_input,
+    const int divisor_override,
+    bool count_include_pad,
+    bool use_divisor) {
+  const scalar_t* top_data = grad_output.data_ptr<scalar_t>();
+  scalar_t* bottom_data = grad_input.data_ptr<scalar_t>();
+
+  auto& queue = at::xpu::getCurrentSYCLQueue();
+  const uint32_t group_size =
+      std::min(static_cast<int>(syclMaxWorkItemsPerEU()), 1024);
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/AveragePool2dKernels.cpp:557]: ditto

File: src/ATen/native/xpu/sycl/AveragePool2dKernels.cpp [hunk 7]
```diff
@@ -0,0 +1,845 @@
...
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/AveragePool2dKernels.cpp:609]: ditto

File: yaml/xpu_functions.yaml (hunk 1 above)
Comment (chunhuanMeng) [reply yaml/xpu_functions.yaml:392]: we will have another pr for this op

**Comment by fengyuan14** [comment]
Aligned with CUDA. Will skip the fail case,
image

**Comment by fengyuan14** [comment]
> Aligned with CUDA. Will skip the fail case, image

`avg_pool1d` is not listed in supported list, so the `dtypes` in test follow CPU definition. Retrieve the case, once `avg_pool1d` is enabled.

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**